code/code/misc/equip.cc

```diff
@@ -1058,8 +1058,8 @@ int TBeing::wear(TObj *o, wearKeyT keyword, TBeing *ch)
               --(*o);
               equipChar(o, HOLD_RIGHT);
               // learn 2h spec as with dual wield
-              if (doesKnowSkill(SKILL_2H_SPEC_DEIKHAN)) {
-                learnFromDoingUnusual(LEARN_UNUSUAL_NORM_LEARN, SKILL_2H_SPEC_DEIKHAN, 60);
+              if (doesKnowSkill(getSkillNum(SKILL_2H_SPEC))) {
+                learnFromDoingUnusual(LEARN_UNUSUAL_NORM_LEARN, SKILL_2H_SPEC, 60);
```

Comment:
I think this is another situation where you'll need to use the result of `getSkillNum(SKILL_2H_SPEC)` in the `learnFromDoingUnusual` call instead of just SKILL_2H_SPEC.

The doesKnowSkill call will successfully make sure that the character knows either SKILL_2H_SPEC (the warrior version) or SKILL_2H_SPEC_DEIKHAN, but will then only actually work for chars who have the warrior version, as coded.